pkgs/development/python-modules/spylls/default.nix

```diff
@@ -0,0 +1,67 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+
+, pythonOlder
+
+, pythonRelaxDepsHook
+
+, poetry-core
+
+, pytest
+}:
+
+buildPythonPackage rec {
+  pname = "spylls";
+  version = "0.1.7";
+  pyproject = true;
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-cEWJLcvTJNNoX2nFp2AGPnj7g5kTckzhgHfPCgyT8iA=";
+  };
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+    --replace-fail poetry.masonry.api poetry.core.masonry.api \
+    --replace-fail "poetry>=" "poetry-core>="
+  '';
+
+  build-system = [
+    poetry-core
+  ];
+
+  nativeBuildInputs = [
+    pytest # hack for fix error: `pytest not installed`
+    pythonRelaxDepsHook
+  ];
+
+  pythonRelaxDeps = [
+    "pytest"
+  ];
+
```

Comment:
```suggestion
```
I can't reproduce such an error.

Reply:
@natsukium this error is only in version 0.1.5
Can you explain how pythonRelaxDeps work, if a package needs another package in version 0.2.0, but the latest version of the package is 0.2.5, do I need to commit version 0.2.0 and then 0.2.5?

Reply:
> if a package needs another package in version 0.2.0, but the latest version of the package is 0.2.5, do I need to commit version 0.2.0 and then 0.2.5?

No. `pythonRelaxDepsHook` simply ignores upstream version constraints.
It does not reference previous versions.